app/models/embedded_ansible_worker/object_management.rb

```diff
@@ -53,4 +54,74 @@ def ensure_host(provider, connection)
       :variables => {'ansible_connection' => "local"}.to_yaml
     ).id
   end
+
+  def ensure_plugin_playbooks_project_seeded(connection)
+    clean_consolidated_plugin_directory
+    copy_plugin_ansible_content
+
+    commit_git_plugin_content
+
+    if project = existing_plugin_playbook_project(connection)
+      update_playbook_project(project)
+    else
+      create_playbook_project(connection)
+    end
+  ensure
+    # we already have 2 copies: one in the gem and one imported into ansible in the project, delete the temporary one
+    clean_consolidated_plugin_directory
+  end
+
+  private
+
+  CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR = Pathname.new("/var/lib/awx_consolidated_source/projects/gem_ansible_content").freeze
+  def clean_consolidated_plugin_directory
+    FileUtils.rm_rf(CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR)
+  end
+
+  def copy_plugin_ansible_content
+    FileUtils.mkdir_p(CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR)
+
+    #TODO: make this a public api via an attr_reader
+    Vmdb::Plugins.instance.instance_variable_get(:@registered_ansible_content).each do |content|
+      FileUtils.cp_r(Dir.glob("#{content.path}/*"), CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR)
+    end
+  end
+
+  def commit_git_plugin_content
+    Dir.chdir(CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR) do
+      require 'rugged'
+      repo = Rugged::Repository.init_at(".")
+      index = repo.index
+      index.add_all("*")
+      index.write
+
+      options              = {}
+      options[:tree]       = index.write_tree(repo)
+      options[:author]     = options[:committer] = { :email => "user@example.com", :name => 'Author', :time => Time.now }
```

Comment:
This is weird. Are the things like email required?

Can we make this use the product name or something? Not sure if this gets exposed in our UI anywhere.

Reply:
Yeah, good question  @syncrou @gmcculloug.  Do we care?

Reply:
@jrafanie - I don't think we care. I would just ask that is consistent, but largely because that seems the right thing to do.

On my testing, I received errors on the commit if the email wasn't set, but that obviously would not be a problem here.

Reply:
I can try not specifying the author and commiter.  On CLI, if there's not author/commit it just warns to STDERR  but still commits.

Reply:
Ugh, rugged requires both author and email for git commits 😞

Reply:
```
irb(main):002:0> EmbeddedAnsibleWorker.new.ensure_plugin_playbooks_project_seeded(Provider.first, EmbeddedAnsible.new)
Rugged::ConfigError: Config value 'user.name' was not found
```